src/Fragments/GoogleTagManager.php

```diff
@@ -25,14 +25,413 @@ public static function addTo(Policy $policy): void
+            'https://adservice.google.com.py',
+            'https://adservice.google.com.qa',
+            'https://adservice.google.ro',
+            'https://adservice.google.ru',
+            'https://adservice.google.rw',
+            'https://adservice.google.com.sa',
+            'https://adservice.google.com.sb',
+            'https://adservice.google.sc',
+            'https://adservice.google.se',
+            'https://adservice.google.com.sg',
+            'https://adservice.google.sh',
+            'https://adservice.google.si',
+            'https://adservice.google.sk',
+            'https://adservice.google.com.sl',
+            'https://adservice.google.sn',
+            'https://adservice.google.so',
+            'https://adservice.google.sm',
+            'https://adservice.google.sr',
+            'https://adservice.google.st',
+            'https://adservice.google.com.sv',
+            'https://adservice.google.td',
+            'https://adservice.google.tg',
+            'https://adservice.google.co.th',
+            'https://adservice.google.com.tj',
+            'https://adservice.google.tl',
+            'https://adservice.google.tm',
+            'https://adservice.google.tn',
+            'https://adservice.google.to',
+            'https://adservice.google.com.tr',
+            'https://adservice.google.tt',
+            'https://adservice.google.com.tw',
+            'https://adservice.google.co.tz',
+            'https://adservice.google.com.ua',
+            'https://adservice.google.co.ug',
+            'https://adservice.google.co.uk',
+            'https://adservice.google.com.uy',
+            'https://adservice.google.co.uz',
+            'https://adservice.google.com.vc',
+            'https://adservice.google.co.ve',
+            'https://adservice.google.vg',
+            'https://adservice.google.co.vi',
+            'https://adservice.google.com.vn',
+            'https://adservice.google.vu',
+            'https://adservice.google.ws',
+            'https://adservice.google.rs',
+            'https://adservice.google.co.za',
+            'https://adservice.google.co.zm',
+            'https://adservice.google.co.zw',
+            'https://adservice.google.cat',
+        ];
+
         $policy
-            ->addDirective(Directive::FRAME, '*.doubleclick.net')
-            ->addDirective(Directive::CONNECT, '*.doubleclick.net')
-            ->addDirective(Directive::IMG, '*.doubleclick.net');
+            ->addDirective(Directive::FRAME,
+                [
+                    'https://*.doubleclick.net',
+                    'https://stats.g.doubleclick.net',
+                    'http://bid.g.doubleclick.net',
+                ]
+            )
+            ->addDirective(Directive::CONNECT, [
+                'https://adservice.google.com',
+                'https://www.google.com',
+                'https://*.doubleclick.net',
+            ])
+            ->addDirective(Directive::IMG, $googleDomains);
```

Comment:
This is pretty large, could you check the header size before/after? I'm a bit worried we'll be sending this down for every request and it might have issues with certain CDNS

Reply:
So I'm getting 12.5kb~ with all the domains vs 2.6KB~ without.

Agreed though I don't like this approach either.

Not sure what to do however as it seems like you either do it or live with tones of reporting violations?

Reply:
@adrhumphreys added new static config to the GTM fragment. Possibly we may want that to live somewhere else though so it can be shared for other Google related fragments? Wasn't sure whats best here 😁